### PR TITLE
This is the new CI-RP stage for the pipline

### PR DESCRIPTION
**Pull request by shubhadapaithankar**
### Which issue this PR addresses:

This PR addresses the issue of inconsistent and error-prone manual build processes by implementing the `make ci `command.

Fixes: [ARO-9563](https://issues.redhat.com/browse/ARO-9563)

### What this PR does / why we need it:

This PR automates the build process by using `make ci` instead of manual make commands, ensuring consistency and reducing the potential for human error in the build steps.

### Test plan for issue:

The pipeline tests to ensure that the make ci command produces the expected build outputs. Additional manual tests were conducted to verify consistency with previous manual build processes.

### Is there any documentation that needs to be updated for this PR?

N/A

### CODE REVIEW

File: .pipelines/ci.yml [hunk 1]
```diff
@@ -28,85 +28,118 @@ resources:
...
-      HOME: $(Agent.BuildDirectory)
+stages:
+- stage: OLD_AND_CRUSTY
+  jobs:
+    - job: Python_Unit_Tests
+      pool:
+        name: 1es-aro-ci-pool
+      variables:
+        HOME: $(Agent.BuildDirectory)
+
```
Comment (SudoBrendan) [inline .pipelines/ci.yml:39]: hey, let's remove these spaces!

```suggestion

```

File: .pipelines/ci.yml [hunk 2]
```diff
@@ -28,85 +28,118 @@ resources:
...
+  
+      steps:
+        - template: ./templates/template-checkout.yml
+        - script: |
+            set -xe
+            make test-python
+            [[ -z "$(git status -s)" ]]
+          displayName: 🧪Run Python Unit Tests
+          target: python
+
```
Comment (SudoBrendan) [inline .pipelines/ci.yml:48]:
```suggestion

```

File: .pipelines/ci.yml [hunk 3]
```diff
@@ -28,85 +28,118 @@ resources:
...
+          target: python
+  
+    - job: Golang_Unit_Tests
+      pool:
+        name: 1es-aro-ci-pool
+      variables:
+        GOCACHE: /tmp/gocache
+      steps:
+        - template: ./templates/template-checkout.yml
+
```
Comment (SudoBrendan) [inline .pipelines/ci.yml:56]:
```suggestion

```

File: .pipelines/ci.yml [hunk 4]
```diff
@@ -28,85 +28,118 @@ resources:
...
+      steps:
+        - template: ./templates/template-checkout.yml
+  
+        - script: |
+            set -xe
+            go version
+            go env
+          displayName: Print Go version & env
+          target: golang
+
```
Comment (SudoBrendan) [inline .pipelines/ci.yml:63]:
```suggestion

```

File: .pipelines/ci.yml [hunk 5]
```diff
@@ -28,85 +28,118 @@ resources:
...
+          displayName: Print Go version & env
+          target: golang
+  
+        - script: |
+            set -xe
+            make generate
+            [[ -z "$(git status -s)" ]]
+          displayName: ⚙️ Run Golang code generate
+          target: golang
+
```
Comment (SudoBrendan) [inline .pipelines/ci.yml:70]:
```suggestion

```

File: .pipelines/ci.yml [hunk 6]
```diff
@@ -28,85 +28,118 @@ resources:
...
+          displayName: ⚙️ Run Golang code generate
+          target: golang
+  
+        - script: |
+            set -xe
+            make build-all
+            [[ -z "$(git status -s)" ]]
+          displayName: 🕵️ Build Golang code
+          target: golang
+
```
Comment (SudoBrendan) [inline .pipelines/ci.yml:77]:
```suggestion

```

File: .pipelines/ci.yml [hunk 7]
```diff
@@ -28,85 +28,118 @@ resources:
...
+            [[ -z "$(git status -s)" ]]
+          displayName: 🕵️ Build Golang code
+          target: golang
+  
+        - script: |
+            set -xe
+            make unit-test-go
+          displayName: 🧪 Run Golang unit tests
+          target: golang
+
```
Comment (SudoBrendan) [inline .pipelines/ci.yml:83]:
```suggestion

```

File: .pipelines/ci.yml [hunk 8]
```diff
@@ -28,85 +28,118 @@ resources:
...
+            make unit-test-go
+          displayName: 🧪 Run Golang unit tests
+          target: golang
+  
+        - task: PublishTestResults@2
+          displayName: 📊 Publish tests results
+          inputs:
+            testResultsFiles: $(System.DefaultWorkingDirectory)/**/report.xml
+          condition: succeededOrFailed()
+
```
Comment (SudoBrendan) [inline .pipelines/ci.yml:89]:
```suggestion

```

File: .pipelines/ci.yml [hunk 9]
```diff
@@ -28,85 +28,118 @@ resources:
...
+            testResultsFiles: $(System.DefaultWorkingDirectory)/**/report.xml
+          condition: succeededOrFailed()
+  
+        - script: |
+            set -xe
+            go run github.com/axw/gocov/gocov@v1.1.0 convert cover.out | go run github.com/AlekSi/gocov-xml@v1.1.0 > coverage.xml
+          displayName: ⚙️ Process Reports
+          condition: succeededOrFailed()
+          target: golang
+
```
Comment (SudoBrendan) [inline .pipelines/ci.yml:96]:
```suggestion

```

File: .pipelines/ci.yml [hunk 10]
```diff
@@ -28,85 +28,118 @@ resources:
...
+  
+        - task: PublishCodeCoverageResults@1
+          displayName: 📈 Publish code coverage
+          inputs:
+            codeCoverageTool: Cobertura
+            summaryFileLocation: $(System.DefaultWorkingDirectory)/**/coverage.xml
+            failIfCoverageEmpty: false
+          condition: succeededOrFailed()
+          target: golang
+
```
Comment (SudoBrendan) [inline .pipelines/ci.yml:105]:
```suggestion

```

Comment (shubhadapaithankar) [reply .pipelines/ci.yml:105]: Removed the space

File: .pipelines/ci.yml [hunk 11]
```diff
@@ -11,106 +10,45 @@ trigger:
...
+  jobs:
+    - job: Build_Lint_Test_RP_Portal
+      pool:
+        name: 1es-aro-ci-pool
+      steps:
+        - template: ./templates/template-checkout.yml
+        - script: |
+            set -xe
+            export NO_CACHE=true
+            if [[ "$BUILD_REASON" == "PullRequest" ]];
```
Comment (SudoBrendan) [inline .pipelines/ci.yml:33]:
```suggestion
            if [[ "##vso[Build.Reason]" == "PullRequest" ]];
```

**Review by SudoBrendan — CHANGES_REQUESTED** [review]

**Review by SudoBrendan — APPROVED** [review]

File: .pipelines/ci.yml [hunk 12]
```diff
@@ -11,106 +10,56 @@ trigger:
...
       - loki/pipeline-swarm
   paths:
     exclude:
       - docs/*
-
-resources:
-  containers:
-    - container: golang
-      image: registry.access.redhat.com/ubi8/go-toolset:1.21.11-1.1720406008
-      options: --user=0
```
Comment (SudoBrendan) [inline .pipelines/ci.yml:27]: we need this back :)

**Review by SudoBrendan — CHANGES_REQUESTED** [review]

**Pull request merged**